app/controllers/posts_controller.rb

```diff
@@ -0,0 +1,73 @@
+class PostsController < ApplicationController
+  before_action :set_post, only: %i[show edit update destroy]
+  before_action :authenticate_user!, except: %i[index show]
```

Comment:
- **_The `before_action` should be use to restrict access to the `:new` and `:create` methods (you can use ApplicationController helper method) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/before_action_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/authentication))_**

Reply:
It's done. Actually it was restricting the new and create before as well, but it has been changed to 'only' from 'except'.